Source/WebKit/UIProcess/SpeechRecognitionPermissionManager.cpp

```diff
@@ -247,7 +257,7 @@ void SpeechRecognitionPermissionManager::requestUserPermission(WebCore::SpeechRe
 void SpeechRecognitionPermissionManager::decideByDefaultAction(const WebCore::SecurityOriginData& origin, CompletionHandler<void(bool)>&& completionHandler)
 {
 #if PLATFORM(COCOA)
-    alertForPermission(protectedPage(), MediaPermissionReason::SpeechRecognition, origin, WTFMove(completionHandler));
```

Comment:
Why don't we null check the page here and call the completion handler with `false` if null? Seems like this could introduce crashes now that we can extend the lifetime of the `SpeechRecognitionPermissionManager` past the page's.

Reply:
Ok. We can add a nullptr check there.